Roles/Neutral/Doppelganger.cs

```diff
@@ -73,6 +77,8 @@ public override void Add(byte playerId)
     // A quick check if a player has been killed by the doppelganger.
     public static bool CheckDoppelVictim(byte playerId) => DoppelVictim.ContainsKey(playerId);
 
+    public static PlayerControl GetDoppelControl(PlayerControl player) => DoppelgangerTarget != null ? DoppelgangerTarget : player;
```

Comment:
btw did you test if storing playercontrol makes it possible to nullify it?

Like if the player leaves, does the reference (Doppelgangertarget in this case) to it's playercontrol also become null?

If not then a better way would to check if main.allplayercontrols.contains(doppelgangertarget), to know if it has left or not.

Reply:
Storing classes with variables in C# keeps the same reference to that instance in all places. So yes, it will be null if they disconnect. (Same applies for lists, dictionaries, etc.)

Reply:
@Ultradragon005 No I haven't tested it but knowing the problems with Dollmaster stored targets yes if the doppelganger does disconnect it will be null, but as you can see if it is null it just Returns the original controller.

Reply:
lol gurge already confirmed my questions